File: 0258-add-digits/0258-add-digits.py
from functools import reduce


class Solution(object):
    def addDigits(self, num):
        string_num = str(num)
        while len(string_num) > 1:
            string_num = str(reduce(lambda x, y: int(x) + int(y), string_num))
        return int(string_num)

File: leetcode/0258-add-digits/0258-add-digits.go
func addDigits(num int) int {
    for {
        sum := sumDigits(num)
        num = sum
        
        if num < 10 {
            return num
        }
    }
}


func sumDigits(num int) int {
    var sum int
    for num > 0 {
        digit := num % 10
        sum += digit
        num /= 10 
    }
    return sum
}
